telemetry/client.go

```diff
@@ -76,58 +77,52 @@ func NewClient(logger *zap.Logger, serverURL string, keyUID string, nodeName str
 		keyUID:             keyUID,
 		nodeName:           nodeName,
 		version:            version,
-		receivedMessagesCh: make(chan ReceivedMessages),
-		receivedEnvelopeCh: make(chan *v2protocol.Envelope),
-		sentEnvelopeCh:     make(chan wakuv2.SentEnvelope),
 		telemetryCh:        make(chan TelemetryRequest),
+		telemetryCacheLock: sync.Mutex{},
+		telemetryCache:     make([]TelemetryRequest, 0),
 		nextId:             0,
+		nextIdLock:         sync.Mutex{},
 		sendPeriod:         10 * time.Second,
 	}
 }
 
-func (c *Client) CollectAndProcessTelemetry(ctx context.Context) {
+func (c *Client) Start(ctx context.Context) {
 	go func() {
+
 		for {
 			select {
-			case receivedMessages := <-c.receivedMessagesCh:
-				c.processAndPushTelemetry(receivedMessages)
-			case receivedEnvelope := <-c.receivedEnvelopeCh:
-				c.processAndPushTelemetry(receivedEnvelope)
-			case sentEnvelope := <-c.sentEnvelopeCh:
-				c.processAndPushTelemetry(sentEnvelope)
+			case telemetryRequest := <-c.telemetryCh:
+				c.telemetryCacheLock.Lock()
+				c.telemetryCache = append(c.telemetryCache, telemetryRequest)
+				c.telemetryCacheLock.Unlock()
 			case <-ctx.Done():
 				return
 			}
 		}
 	}()
-}
-
-func (c *Client) Start(ctx context.Context) {
-	go c.CollectAndProcessTelemetry(ctx)
 	go func() {
 		ticker := time.NewTicker(c.sendPeriod)
 		defer ticker.Stop()
 
 		for {
 			select {
 			case <-ticker.C:
-				var telemetryRequests []TelemetryRequest
-				collecting := true
-				for collecting {
-					select {
-					case telemetryRequest := <-c.telemetryCh:
-						telemetryRequests = append(telemetryRequests, telemetryRequest)
-					default:
-						collecting = false
-					}
-				}
-				if len(telemetryRequests) > 0 {
+				c.telemetryCacheLock.Lock()
+				if len(c.telemetryCache) > 1 {
+					telemetryRequests := make([]TelemetryRequest, len(c.telemetryCache))
+					copy(telemetryRequests, c.telemetryCache)
+					c.telemetryCache = make([]TelemetryRequest, 0)
+					c.telemetryCacheLock.Unlock()
+
 					c.pushTelemetryRequest(telemetryRequests)
+				} else {
+					c.telemetryCacheLock.Unlock()
 				}
```

Comment:
To avoid having to call `Unlock` in both the `if` and `else`, perhaps this could be refactored to look like this?:
```go
c.telemetryCacheLock.Lock()
telemetryRequests := make([]TelemetryRequest, len(c.telemetryCache))
copy(telemetryRequests, c.telemetryCache)
c.telemetryCache = nil
c.telemetryCacheLock.Unlock()

if len(telemetryRequests) > 0 {
    c.pushTelemetryRequest(telemetryRequests)
}
```